src/Chainweb/MerkleLogHash.hs

```diff
@@ -83,7 +83,7 @@ newtype MerkleLogHash = MerkleLogHash (MerkleRoot (HashAlg ChainwebHashTag))
 -- | Smart constructor
 --
 merkleLogHash :: MonadThrow m => B.ByteString -> m MerkleLogHash
-merkleLogHash = fmap MerkleLogHash . decodeMerkleRoot
+merkleLogHash = fmap MerkleLogHash . decodeMerkleRoot . B.copy
```

Comment:
This is called internally

```Haskell
-- | Decode a 'MerkleHash' from bytes.
--
decodeMerkleHash
    :: forall a b m
    . MonadThrow m
    => HashAlgorithm a
    => BA.ByteArrayAccess b
    => b
    -> m (MerkleHash a)
decodeMerkleHash b
    | BA.length b /= hashSize @a = throwM e
    | otherwise = return $ MerkleHash $ BA.convert b
  where
    e = EncodingSizeException "MerkleHash"
        (Expected (hashSize @a @Int))
        (Actual (BA.length b))
{-# INLINE decodeMerkleHash #-}
```

`MerkleHash` is a new type over `BA.Bytes`. `BA.convert` internally uses ` copyByteArrayToPtr` from the `ByteArrayAccess` class. The default implementation uses C `memcpy`. Specialized implementation should still respect the "copy" in the name of the function. So we should be fine.